addons/html_editor/static/src/main/media/media_plugin.js

```diff
@@ -131,4 +134,165 @@ export class MediaPlugin extends Plugin {
             ...params,
         });
     }
+
+    async savePendingImages() {
+        const editableEl = this.editable;
+        const { resModel, resId } = this.config.recordInfo;
+        // When saving a webp, o_b64_image_to_save is turned into
+        // o_modified_image_to_save by saveB64Image to request the saving
+        // of the pre-converted webp resizes and all the equivalent jpgs.
+        const b64Proms = [...editableEl.querySelectorAll(".o_b64_image_to_save")].map(
+            async (el) => {
+                const dirtyEditable = el.closest(".o_dirty");
+                if (dirtyEditable && dirtyEditable !== editableEl) {
+                    // Do nothing as there is an editable element closer to the
+                    // image that will perform the `saveB64Image()` call with
+                    // the correct "resModel" and "resId" parameters.
+                    return;
+                }
+                await this.saveB64Image(el, resModel, resId);
+            }
+        );
+        const modifiedProms = [...editableEl.querySelectorAll(".o_modified_image_to_save")].map(
+            async (el) => {
+                const dirtyEditable = el.closest(".o_dirty");
+                if (dirtyEditable && dirtyEditable !== editableEl) {
+                    // Do nothing as there is an editable element closer to the
+                    // image that will perform the `saveModifiedImage()` call
+                    // with the correct "resModel" and "resId" parameters.
+                    return;
+                }
+                await this.saveModifiedImage(el, resModel, resId);
+            }
+        );
+        const proms = [...b64Proms, ...modifiedProms];
+        const hasChange = !!proms.length;
+        if (hasChange) {
+            await Promise.all(proms);
+        }
+        return hasChange;
+    }
+
+    /**
+     * Saves a base64 encoded image as an attachment.
+     * Relies on saveModifiedImage being called after it for webp.
+     *
+     * @private
+     * @param {Element} el
+     * @param {string} resModel
+     * @param {number} resId
+     */
+    async saveB64Image(el, resModel, resId) {
```

Comment:
To be discussed.
2 RPCs per image... requesting an access token afterwards... All this could be simplified and optimized with a smarter route (not to mention multiple images could be sent as a batch rather than individually). The controller is complicated because of some website needs. It could be made better/simpler, and they could extend the controller with their own patches/overrides.
Same for this plugin. IMO it should not contain website specific stuff, they should have their inherited plugin instead.
I know it's way out of scope for this PR, but I'd like to bring up the discussion now.